hook-worker/src/worker.rs

```diff
@@ -249,10 +249,10 @@ async fn process_webhook_job<W: WebhookJob>(
 
     match send_result {
         Ok(_) => {
-            webhook_job
-                .complete()
-                .await
-                .map_err(|error| WorkerError::PgJobError(error.to_string()))?;
+            webhook_job.complete().await.map_err(|error| {
+                metrics::counter!("webhook_jobs_database_error", &labels).increment(1);
+                WorkerError::PgJobError(error.to_string())
```

Comment:
It's minor, but I think we should have a different enum entry like `DatabaseError` too. Just helps for grep in the future.

Reply:
Doesn't have to be for this PR btw.

Reply:
Hmm, I was thinking in my head things like `complete()` returned a `sqlx::Error` but I guess those are wrapped internally:

https://github.com/PostHog/rusty-hook/blob/ead1bf28ef8d092848cceedf8e2ae8867b02276d/hook-common/src/pgqueue.rs#L30-L37

It's a little weird that `RetryInvalidError` will get treated as a DB error here. Although I guess that should never happen, since we're just calling `complete`/`fail`, it feels like the types could help us better here somehow.

Reply:
Yeah, I agree we should differentiate between `RetryInvalidError` and database errors. `RetryInvalidError` is the only special error as it returns the job itself to be retried, so I'd split it off this enum and rename the enum to `DatabaseError` or something similar.

Reply:
Happy to update this PR, will need a bit of time as I have some support tickets to work through.

Reply:
> Happy to update this PR, will need a bit of time as I have some support tickets to work through.

No big deal, it was mostly just a thought. Feel free to merge this if you want too, either way.